src/java.desktop/share/classes/com/sun/java/swing/SwingUtilities3.java

```diff
@@ -135,4 +143,72 @@ public static RepaintManager getDelegateRepaintManager(Component
         }
         return delegate;
     }
+
+    @FunctionalInterface
+    public interface PaintInterface {
+        void paintUnscaledBorder(Component c, Graphics g, int x, int y,
+                                 int w, int h, double scale, int strokeWidth);
+    }
+
+    public static void paintBorder(Component c, Graphics g, int x, int y,
+                            int w, int h, PaintInterface paintFunction) {
```

Comment:
`painter` could be a better name than `paintFunction`.

Even though a method reference is used as the parameter, here it's still an object that implements the specified interface.